### PR TITLE
Add configuration options for Dynamic Content Assembly (DCA) defaults

### LINKED ISSUE #46: Option to configure Dynamic Content Assembly (DCA) defaults?
As a feature request, would it be possible to configure the default for the `dca` handling of `<esi:include>` found within streams?

This library now mirrors Akamai ESI 5.0, which [supports `dca` defaults via `Edge-Control: dca=esi` response header](https://techdocs.akamai.com/property-mgr/docs/esi-edge-side-includes), thus allows for control of this behaviour outside of markup.

The [fastly JS library for ESI](https://github.com/fastly/compute-js-esi) also [defaults to recursive inclusion](https://github.com/fastly/compute-js-esi?tab=readme-ov-file#max-inclusion-depth), and supports configurable depth - which might also be nice to configure here, too.

Once again, thanks for open sourcing this!

### PR DESCRIPTION
**Pull request by vagetman**
## Problem

There is currently no way to control the default DCA (Dynamic Content Assembly) behaviour for `<esi:include>` and `<esi:eval>` tags — either globally across all fragments, or on a per-fragment basis via response headers. Users who want Akamai-style "ESI-by-default" processing must add `dca="esi"` to every tag. There is also no depth limit enforcement, no `Edge-Control` header support, and no way for child fragments to inherit a parent's DCA mode.

Fixes fastly/esi#46

## Changes

Adds four new configuration options to `Configuration`:

- **`default_dca`** (`DcaMode`, default: `None`) — Sets the global default DCA mode for includes/evals without an explicit `dca` attribute. Set to `DcaMode::Esi` to enable Akamai-style behaviour where all fragments are ESI-processed by default.

- **`max_include_depth`** (`usize`, default: `15`) — Enforces a maximum nesting depth for ESI includes/evals. The EdgeSuite spec defaults to 15 levels. When the limit is reached, fragment content is silently omitted.

- **`enable_edge_control`** (`bool`, default: `false`) — Enables parsing of `Edge-Control` response headers for per-fragment DCA directives (e.g. `Edge-Control: dca=esi`). Matches Akamai's "Enable Through Response Headers" property setting.

- **`inherit_parent_dca`** (`bool`, default: `false`) — When enabled, child fragments inside an explicit `dca="esi"` subtree inherit ESI processing without needing their own attribute. Top-level includes still use `default_dca`.

Precedence (highest wins): tag attribute → Edge-Control header → inherited parent DCA → `default_dca`.

Also makes `DcaMode` a public export and changes the internal `dca` field from `DcaMode` to `Option<DcaMode>` to distinguish "not specified" from "explicitly set to none".

## Usage

```rust
let config = esi::Configuration::default()
    .with_default_dca(esi::DcaMode::Esi)
    .with_edge_control(true)
    .with_inherit_parent_dca(true)
    .with_max_include_depth(5);

let processor = esi::Processor::new(Some(req), config);
```

## Tests

Comprehensive test suite in `dca_tests.rs` covering all four features (23 tests total), including precedence interactions between config, tag attributes, Edge-Control headers, and inheritance.

## Example

New `esi_dca_example` demonstrating all four DCA configuration options.

### CODE REVIEW

File: examples/esi_dca_example/fastly.toml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# This file describes a Fastly Compute package. To learn more visit:
+# https://developer.fastly.com/reference/fastly-toml/
+
+authors = ["kblanks@fastly.com"]
```
Comment (kailan) [inline examples/esi_dca_example/fastly.toml]:
```suggestion
authors = ["vadim@fastly.com"]
```

:)

**Review by kailan — DISMISSED** [review]
This looks really good. Thanks for the well-defined commits!

**Review by kailan — APPROVED** [review]
